tests/test_google_analytics_provider.py

```diff
@@ -0,0 +1,59 @@
+from nose.tools import (
+    eq_,
+)
+from api.config import (
+    Configuration,
+    temp_config,
+)
+from core.analytics import Analytics
+from api.google_analytics_provider import GoogleAnalyticsProvider
+from . import DatabaseTest
+from core.model import CirculationEvent
+import urlparse
+import datetime
+
+class MockGoogleAnalyticsProvider(GoogleAnalyticsProvider):
```

Comment:
Again, in this sort of mock class it's frequently useful to see a complete log of things that passed through it, not just the most recent thing.